keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/optimizer/Variable.java

```diff
@@ -1,10 +1,19 @@
 package io.improbable.keanu.algorithms.variational.optimizer;
 
-public interface Variable<T> {
+import io.improbable.keanu.network.VariableState;
+import io.improbable.keanu.vertices.Samplable;
+
+public interface Variable<VALUE, STATE extends VariableState> extends Samplable<VALUE> {
```

Comment:
NB: the purpose of `VariableState` is to abstract the concept of state. For a `Vertex` its state comprises two things: `value` and `isObserved`.